overlord/configstate/transaction.go

```diff
@@ -85,78 +83,101 @@ func (t *Transaction) Set(snapName, key string, value interface{}) error {
 	config[key] = &raw
 
 	// Put that config into the write cache
-	t.writeCache[snapName] = config
+	t.changes[snapName] = config
 
 	return nil
 }
 
-// Get extracts the stored value associated with the provided key for the snap's
-// config into the value parameter. Note that this query is run against the
-// cached copy of the config; you won't see updates from other transactions.
-func (t *Transaction) Get(snapName, key string, value interface{}) error {
-	// Extract the config for this specific snap. Check the cache first; if it's
-	// not there, check the config from the state
-	t.writeCacheMutex.RLock()
-	defer t.writeCacheMutex.RUnlock()
-	if err := t.get(t.writeCache, snapName, key, value); err != nil {
-		t.configMutex.RLock()
-		defer t.configMutex.RUnlock()
-		return t.get(t.config, snapName, key, value)
+// Get unmarshals into result the cached value of the provided snap's configuration key.
+// If the key does not exist, an error of type *NoOptionError is returned.
+//
+// Transactions do not see updates from the current state or from other transactions.
+func (t *Transaction) Get(snapName, key string, result interface{}) error {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+	err := t.get(t.changes, snapName, key, result)
+	if _, ok := err.(*NoOptionError); ok {
+		err = t.get(t.pristine, snapName, key, result)
 	}
+	return err
+}
 
+// GetMaybe unmarshals into result the cached value of the provided snap's configuration key.
+// If the key does not exist, no error is returned.
+//
+// Transactions do not see updates from the current state or from other transactions.
+func (t *Transaction) GetMaybe(snapName, key string, result interface{}) error {
```

Comment:
This isn't used in this PR. What is the use-case behind it? Why will callers care whether or not the error was because the key wasn't found or due to something else?